src/particles/ImpactXParticleContainer.cpp

```diff
@@ -167,6 +167,83 @@ namespace impactx
                 particle_tile, pinned_tile, 0, old_np, pinned_tile.numParticles());
     }
 
+    void
+    ImpactXParticleContainer::AddNParticles (int lev,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & x,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & y,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & t,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & px,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & py,
+                                             amrex::Gpu::DeviceVector<amrex::ParticleReal> const & pt,
+                                             amrex::ParticleReal qm,
+                                             amrex::ParticleReal bchchg)
+    {
+        BL_PROFILE("ImpactX::AddNParticles");
+
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(lev == 0, "AddNParticles: only lev=0 is supported yet.");
```

Comment:
ImpactX support MR now. I wonder if `lev!=0` is generally needed or if we want to remove the parameter altogether.

I assume that on `Redistribute` we automatically move the particles in the corresponding level at the end of `AddNParticles`?